src/test/java/uk/nhs/prm/deduction/e2e/tests/EhrOutE2ETests.java

```diff
@@ -0,0 +1,247 @@
+        String nhsNumberForTestPatient = "9727018440";
+        String previousGpForTestPatient = "M85019";
+        String asidCodeForTestPatient = "200000000149";
+        LOGGER.info(" ===============  outboundConversationId: {}", outboundConversationId);
+
+        SimpleAmqpQueue inboundQueueFromMhs = new SimpleAmqpQueue(config);
+
+        String smallEhr = getSmallEhrWithoutLinebreaks(inboundConversationId.toUpperCase(), smallEhrMessageId);
+        String ehrRequest = getEhrRequest(nhsNumberForTestPatient, previousGpForTestPatient, asidCodeForTestPatient, outboundConversationId);
+
+        // When
+        // change transfer db status to ACTION:EHR_REQUEST_SENT before putting on inbound queue
+        // Put the patient into inboundQueueFromMhs as a UK05 message
+
+        addRecordToTrackerDb(trackerDb, inboundConversationId, "", nhsNumberForTestPatient, previousGpForTestPatient, "ACTION:EHR_REQUEST_SENT");
+        inboundQueueFromMhs.sendMessage(smallEhr, inboundConversationId);
+
+        LOGGER.info("conversationIdExists: {}",trackerDb.conversationIdExists(inboundConversationId));
+        String status = trackerDb.waitForStatusMatching(inboundConversationId, "ACTION:EHR_TRANSFER_TO_REPO_COMPLETE");
+        LOGGER.info("tracker db status: {}", status);
+
+        // Send an EHR request from inboundQueueFromMhs
+        inboundQueueFromMhs.sendMessage(ehrRequest, outboundConversationId);
+
+        // Then
+        SqsMessage gp2gpMessage = gp2gpMessengerQueue.getMessageContaining(outboundConversationId);
+
+        String gp2gpMessengerPayload = getPayloadOptional(gp2gpMessage.body()).orElseThrow();
+        String smallEhrPayload = getPayloadOptional(smallEhr).orElseThrow();
+        LOGGER.info("Payload from gp2gpMessenger: {}", gp2gpMessengerPayload);
+        LOGGER.info("Payload from smallEhr: {}", smallEhrPayload);
+
+        assertThat(gp2gpMessage).isNotNull();
+        assertTrue(gp2gpMessage.contains("RCMR_IN030000UK06"));
+        assertTrue(gp2gpMessengerPayload.contains(nhsNumberForTestPatient));
+
+        // clear up the queue after test in order not to interfere with other tests
+        gp2gpMessengerQueue.deleteMessage(gp2gpMessage);
+    }
+
+    @Test
+    void shouldPutALargeEHROntoRepoAndSendEHRToMHSOutboundWhenReceivingRequestFromGP() {
+        // given
+        String inboundConversationId = UUID.randomUUID().toString();
+        String outboundConversationId = UUID.randomUUID().toString();
+        LOGGER.info(" ===============  outboundConversationId: {}", outboundConversationId);
+
+        String largeEhrCoreMessageId = UUID.randomUUID().toString();
+        String fragment1MessageId = UUID.randomUUID().toString();
+        String fragment2MessageId = UUID.randomUUID().toString();
+
+        String nhsNumberForTestPatient = "9727018157";
+        String previousGpForTestPatient = "N82668";
+        String newGpForTestPatient = "M85019";
```

Comment:
Just popping this in as a note to the discussion we just had - would be beneficial to identify areas that are repeated throughout tests and abstract them out into small POJO's - the builder pattern would be great here!